packages/ngtools/webpack/src/compiler_host.ts

```diff
@@ -147,28 +147,22 @@ export class WebpackCompilerHost implements ts.CompilerHost {
     const filePath = this.resolve(fileName);
 
     try {
+      let content: ArrayBuffer;
       if (this._memoryHost.isFile(filePath)) {
-        return virtualFs.fileBufferToString(this._memoryHost.read(filePath));
+        content = this._memoryHost.read(filePath);
       } else {
-        const content = this._syncHost.read(filePath);
-
-        return virtualFs.fileBufferToString(content);
+        content = this._syncHost.read(filePath);
       }
+
+      // strip BOM
+      return virtualFs.fileBufferToString(content).replace(/^\uFEFF/, '');
     } catch {
       return undefined;
     }
   }
 
   readFileBuffer(fileName: string): Buffer {
-    const filePath = this.resolve(fileName);
-
-    if (this._memoryHost.isFile(filePath)) {
-      return Buffer.from(this._memoryHost.read(filePath));
-    } else {
-      const content = this._syncHost.read(filePath);
-
-      return Buffer.from(content);
-    }
+    return Buffer.from(this.readFile(fileName) || '');
```

Comment:
This is backwards.  The Buffer should be the raw data from the file.  The text content is based on this and may contain a _header_ (i.e., the BOM).  Nothing should be altered when getting the raw data.  It's completely valid for a binary file to contain the BOM values.